### ISSUE TITLE
Feed not working

### ISSUE THREAD

**Issue by w9925** [j-fbriere/squawker#151]
Since today the feed is no longer working.

**Comment by brnwlshubh** [comment]
I think it is fixed now https://github.com/j-fbriere/squawker/commit/f1c6242022377df92fde9cc42c824307e7328128

**Comment by appcreatorguy** [comment]
Can confirm that this latest build has not fixed the issue.

**Comment by zer0fl0w** [comment]
+1

**Comment by Gotchfutchian** [comment]
Latest commit build does not work for me either.

**Comment by j-fbriere** [comment]
The new version [v3.6.0](https://github.com/j-fbriere/squawker/releases/tag/v3.6.0) has an option to switch the type of Twitter/X requests for feeds to "enhanced requests", which might resolve those types of problems.
This setting is activated by default.
Note though that this new type of request as a lower rate limit (50 requests per 15 min instead of 150).
The option to switch is found under 'Settings/General/X API' subsection.

**Comment by GeographicCone** [comment]
Same for me since yesterday. I've just tried with 3.6.0 and it didn't resolve the problem, regardless of whether the _Enhanced Requests_ are off or on (the new default). The error message now is "Unable to create the guest account" similar to #72.

Since the previous version was from F-Droid, had to uninstall first due to signature conflict. If it's caused by the fact that an account was already created today as suggested [in this comment](https://github.com/j-fbriere/squawker/issues/72#issuecomment-1716947131) and the issue resolves itself with _Enhanced Requests_ on tomorrow, I'll update my comment here but for now, adding this for completeness.

**Thank you for all your work on the app, I really appreciate it.**

**Comment by j-fbriere** [comment]
@w9925 Yes as of today it seems that the "old" API requests for the feeds returns empty data. Maybe it's only temporary. I will do follow-ups about it.

**Comment by GeographicCone** [comment]
@j-fbriere and @w9925 Thank you for your comments!

I was able to get it to work by manually generating a guest account using the [Nitter Wiki shell script](https://github.com/zedeus/nitter/wiki/Guest-Account-Branch-Deployment#guest-accounts) modified to use Tor's SOCKS proxy by adding `-x socks5://127.0.0.1:9150/` to `curl(1)`.

I then pulled `/data/data/org.ca.squawker/databases/squawker.db` from the device and manually inserted the obtained (`screen_name`, `oauth_token`, `oauth_token_secret`) into the `guest_account` table within the SQLite database, pushed the file back to the device, and voilà: it's working.

**Comment by j-fbriere** [comment]
@GeographicCone thank you for your effort and the information provided. 
I wonder if Tor is installed on the device, X guest account could be generated through its proxy by Squawker, given the appropriate proxy's configuration. I'll look into it.
Also, there is a new [feature request](https://github.com/j-fbriere/squawker/issues/152) to add custom list of guest accounts with a file that I will implement. It should be easier that way to add new accounts :-)

**Comment by zer0fl0w** [comment]
can confirm v3.6.0 working again.

thanks as always.

fyi: search is not working anymore. 
probably expected. but therefore also hashtags are  not working anymore.

**Comment by j-fbriere** [comment]
@zer0fl0w Yes as of today the search , who uses the "old" API, is not working.
I'm working to implement the "enhanced" API version of the search, which is working but have a lower rate limit of 50 requests per 15 minutes (like the feed).

**Comment by holty07** [comment]
Just wanted to chime in and say that once I updated to this new version I was having the "unable to create guest account" error. I turned on my VPN to a new location and bam my feed populated. Not sure if that is helpful at all.

**Comment by anyone00** [comment]
I did not have any rate-limiting before but now with the new "enhanced requests" turned on I get "The request TweetDetails has reached a it's limit. Please wait 12 hours." Was this error added recently? I don't remember it occurring before

SearchTimelines also has a similar error message :frowning_face:

**Comment by Cherkassiy** [comment]
Feed is still not working on my end, tried reinstalling and VPN too, the guest token creation fails. https://i.imgur.com/C8Bk6GK.png

**Comment by zer0fl0w** [comment]
> I did not have any rate-limiting before but now with the new "enhanced requests" turned on I get "The request TweetDetails has reached a it's limit. Please wait 12 hours." Was this error added recently? I don't remember it occurring before
> 
> SearchTimelines also has a similar error message ☹️

new api is heavily rate-limiting. see changelogs of v3.6.0. not squawkers fault. it is technical limitation on twitters side.

**Comment by anyone00** [comment]
@j-fbriere does enabling the "Keep feeds offset" use more requests somehow? or is it handled locally?

**Comment by Hasan5750** [comment]
> @j-fbriere and @w9925 Thank you for your comments!
> 
> I was able to get it to work by manually generating a guest account using the [Nitter Wiki shell script](https://github.com/zedeus/nitter/wiki/Guest-Account-Branch-Deployment#guest-accounts) modified to use Tor's SOCKS proxy by adding `-x socks5://127.0.0.1:9150/` to `curl(1)`.
> 
> I then pulled `/data/data/org.ca.squawker/databases/squawker.db` from the device and manually inserted the obtained (`screen_name`, `oauth_token`, `oauth_token_secret`) into the `guest_account` table within the SQLite database, pushed the file back to the device, and voilà: it's working.

Add consumer key and consumer secret options to sqlite file in the databases, no login options so as not to break the rules,

**Comment by GeographicCone** [comment]
@j-fbriere

> I'll look into it.

Just for the record, I wasn't asking you to do anything. The issue I had was solved and for me the app is great as is. I really appreciate your keeping up with the incessant API changes. Thank you!

@holty07

> I was having the "unable to create guest account" error. I turned on my VPN to a new location and bam my feed populated.

Yeah, that's a more straightforward solution. I only used Tor as a last resort after I discovered my ISP no longer allocates a fresh IP address on every reconnect and connecting via mobile data wouldn't generate a new guest account either (probably because they use a heavily-shared pool of IP addresses).

Just in case anyone ever needs it though, here's a somewhat streamlined shell script to obtain a new guest account via Tor running on `localhost` based on [the original](https://github.com/zedeus/nitter/wiki/Guest-Account-Branch-Deployment#guest-accounts):

````shell
#!/bin/sh

agent="TwitterAndroid/10.10.0"
api_url="https://api.twitter.com/1.1"
api_url_activate="${api_url}/guest/activate.json"
api_url_flow="${api_url}/onboarding/task.json?flow_name=welcome"
api_url_task="${api_url}/onboarding/task.json"
auth_token="AAAAAAAAAAAAAAAAAAAAAFXzAwAAAAAAMHCxpeSDG1gLNLghVe8d74hl6k4%3DRUMF4xAQLsbeBhTSRrCiQpJtxoGWeyHrDb5te2jpGskWDFW82F"
params="-A \"${agent}\" -s -x socks5://127.0.0.1:9150/ -XPOST"

guest_token=`curl ${params} "${api_url_activate}" \
    -H "Authorization: Bearer ${auth_token}" \
    | jq -r .guest_token`

flow_token=`curl ${params} "${api_url_flow}" \
    -H "Authorization: Bearer ${auth_token}" \
    -H "Content-Type: application/json" \
    -H "X-Guest-Token: ${guest_token}" \
    -d '{"flow_token":null,"input_flow_data":{"flow_context":{"start_location":{"location":"splash_screen"}}}}' \
    | jq -r .flow_token`

curl ${params} "${api_url_task}" \
    -H "Authorization: Bearer ${auth_token}" \
    -H "Content-Type: application/json" \
    -H "X-Guest-Token: ${guest_token}" \
    -d "{\"flow_token\":\"${flow_token}\",\"subtask_inputs\":[{\"open_link\":{\"link\":\"next_link\"},\"subtask_id\":\"NextTaskOpenLink\"}]}" \
    | jq -c -r '.subtasks[0]|if(.open_account) then {screen_name: .open_account.user.screen_name, oauth_token: .open_account.oauth_token, oauth_token_secret: .open_account.oauth_token_secret} else empty end'
````

@Hasan5750

> So as not to break the rules

What rules? What do you mean?

I have permission from the owner of my phone (me) to do anything I want with it.

**Comment by BJ24** [comment]
I'm sorry, I too am having this issue but am unable to find V 3.6.0 
Am I to compile this myself?
Are there instructions to do so?

**Comment by brnwlshubh** [comment]
Single profile is easily accessible but not from Groups.

**Comment by zer0fl0w** [comment]
> I'm sorry, I too am having this issue but am unable to find V 3.6.0 Am I to compile this myself? Are there instructions to do so?

[v3.6.0](https://github.com/j-fbriere/squawker/releases/download/v3.6.0/squawker-v3.6.0_arm64-v8a.apk)

**Comment by anyone00** [comment]
Is this a new limitation? nitter has been working fine as before.
Maybe they use a different technique or they use many guest accounts.

**Comment by j-fbriere** [comment]
Nitter uses thousands of guest accounts to overcome the very restrictive limitation of the "enhanced" requests that it's also using. 
Nitter generates those guest accounts with the help of some proxies that can provide multiple IP addresses.
Remember that the Twitter/X rule is typically one guest account creation per IP address per day.
This type of strategy is hard to be applied with Squawker without any proxy, so it will be able to create only one guest account per IP address per day.

**Comment by anyone00** [comment]
I guess the best solution moving forward is to have a list of guest accounts that you generate on your own and insert into the app. wdyt?

**Comment by j-fbriere** [comment]
The new version [v3.6.1](https://github.com/j-fbriere/squawker/releases/tag/v3.6.1) is creating a new Twitter/X guest account each day (Twitter/X forbids to create more than one per day), so that after some time the pool of Twitter/X guest accounts will be large enough to avoid any rate limit problem for the enhanced requests.

**Comment by bowlingotter** [comment]
3.6.2 worked on first open but the feed stopped working on second open, with or without enhanced

**Comment by j-fbriere** [comment]
@bowlingotter Do you have an error?
Also, can you see the tweets of a subscription?
Finally, can you do a search?

**Comment by beerninja** [comment]
3.6.2 fixed the search but feed is blank still.  I am able to view all tweets under individual subscriptions.

**Comment by j-fbriere** [comment]
@beerninja This is strange because the "enhanced" search uses the same type of Twitter/X API request than the "enhanced" feed.
So if one works the other one should also work.
Make sure that the "enhanced" feed is activated in the settings under the General / X API subsection.

**Comment by beerninja** [comment]
> @beerninja This is strange because the "enhanced" search uses the same type of Twitter/X API request than the "enhanced" feed. So if one works the other one should also work. Make sure that the "enhanced" feed is activated in the settings under the General / X API subsection.

Yep it's checked.  I tried unchecking it and restarting app and checking it again and restarting app but feed still blank unfortunately.  Should I backup my subs clear cache/data?  My android version is ancient 8.0 so not sure if that affects anything.

**Comment by j-fbriere** [comment]
Hard to tell what is the problem.
I just tested on an Android 8.0 emulator and everything works.
It's not a real device but still it is an Android 8.0.

Which device are you using?

You can try to uninstall/install (which will clear the cache/data) but it's important before uninstalling to export all your subs/groups/members AND guest accounts, so that you'll import it after installing.

**Comment by Hasan5750** [comment]
Why flutter? Why not react native?

**Comment by autogris** [comment]
With both feed and search with enhanced options activated, my feed is blank. 
Accessing user _tweets_ and _tweet and replies_ through _subscriptions_ gives the error:

`FormatException: Unexpected end of input at character 1)`

_Followers_ and _following_ works as expected, though. 
Search is also working as expected.

I've tried clearing the cache data, but it didn't solved the problem.

I'm on version v3.6.2+300012832 (Android 8.1.0)

**Comment by j-fbriere** [comment]
> Accessing user tweets and tweet and replies through subscriptions gives the error:
>
> `FormatException: Unexpected end of input at character 1)`

@autogris Version [v3.6.3](https://github.com/j-fbriere/squawker/releases/tag/v3.6.3) fixes this issue.

**Comment by autogris** [comment]
Just a note that, after installing 3.6.3, it appeared to be broken by showing empty feed but working user profiles, and after a restart loaded some feed tweets and several "out of range" errors. **BUT**, it all fixed up with **clearing the cache**. Everything seems to be working now, so far. Thanks!
